### ISSUE TITLE
Implementation of shader merge / my documentation of it had issues

### ISSUE THREAD

**Issue by ultimatespirit** [cozyGalvinism/reshader#5]
(This is mainly copied from what I posted as a review on the original merge to handle issue #2, the analysis at the end has been expanded a bit)

https://github.com/cozyGalvinism/reshader/blob/6ebffe660a4c6e54ee6d15d4d36b4ead9018d5b0/src/lib/lib.rs#L147

We run into an ordering issue here between `crosire/reshade-shaders` and `CeeJayDK/SweetFX`, where if one repo had overlapping files that were not exactly the same then the order in which we add them matters. It stands to reason that it's likely for `crosire/reshade-shaders` to be the base directory over which things get overwritten, if needed. Currently I believe the ordering of repos in `SHADER_REPOSITORIES` makes it so `SweetFX` would get overwritten.

And indeed, if you run the following (assuming all pulled repositories are in the CWD at start)....
```
$ cd SweetFX
$ shasum Shaders/* > ../SweetFX-Shaders-SHAsums.txt
$ cd ../reshade-shaders
$ shasum -qc ../SweetFX-Shaders-SHAsums.txt --ignore-missing
Shaders/FXAA.fx: FAILED
Shaders/Nostalgia.fx: FAILED
Shaders/SMAA.fx: FAILED
shasum: WARNING: 3 computed checksums did NOT match
```
So at time of writing three files in `SweetFX` that are also in `crosire/reshade-shaders` on the `master` branch do *not* match.

I checked the SHAs versus a copy of `reshade-shaders` from a fresh install on Windows that a friend provided me and `SweetFX` shaders match perfectly, whereas `crosire/reshade-shaders` did not match for 6 files (the 3 noted above that conflict between SweetFX and itself, as well as 3 additional files that are newer in the fresh install than what appears to be currently on `master` in `crosire/reshade-shaders`. These files appear to be from the `slim` branch.).

For what it's worth, there is _no_ overlap currently between `SweetFX` and the `slim` branch, and it seems none of the files on `master` that are not in `slim` are in the fresh install. So I think this may be a case of the infrastructure having changed since the time of the shell script my original suggestion was based off of. Indeed that script actually didn't even make the subdirectories currently seen, it just shoved everything into one flat `Shaders` directory, though it did not overwrite anything.

Tl;dr it seems we need to use the `slim` branch to conform most closely to what a pristine installation does, and regardless we want `crosire/reshade-shaders` to go first / get overwritten when a conflict emerges. If we did want to ship the extra shaders `master` has over `slim` it would require some finangling since `slim` is definitely newer than `master` is currently.

Also having said all that... it seems this file is our canonical information sheet on what to install and how: https://github.com/crosire/reshade-shaders/blob/list/EffectPackages.ini

From the `list` branch of `crosire/reshade-shaders`.

So... that's an oops on my part sorry. Parsing that file and just following it is likely to be the best path forward. This also explains why I had heard things like "installing all shaders is a bad idea because <some repository names that were not in the original shell script> conflicts with these presets". Turns out there are more options the installer lists. Choosing to install any or all of them probably should be emulated directly with a select dialogue the user goes through (or specifies at the CLI, though that gets increasingly cumbersome). The default install appears to be solely `crosire/reshade-shaders` (slim branch) and `SweetFX` (as marked by the "Enabled" flag).

**Comment by cozyGalvinism** [comment]
Hmm... So we'd need multiple install options for shaders. One which really just installs the bare minimum and one which allows for a selection of repositories.

If someone just wants to use presets, they'd be better off using the GShade shaders and presets anyways, right? Still something we might want to provide using ReShade stuff in the future.

**Comment by cozyGalvinism** [comment]
Implemented in cd1e442, needs some testing though
